velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -215,11 +215,10 @@ class S3WriteFile::Impl {
       const std::string& path,
       Aws::S3::S3Client* client,
       memory::MemoryPool* pool)
-      : client_(client) {
+      : client_(client), pool_(pool) {
     VELOX_CHECK_NOT_NULL(client);
     VELOX_CHECK_NOT_NULL(pool);
     getBucketAndKeyFromS3Path(path, bucket_, key_);
-    pool_ = pool->addLeafChild(".s3writefile");
```

Comment:
@xiaoxmeng The test failure showed that we pass a leaf pool to the FileSink options.
https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/HiveDataSink.cpp#L505
We cannot add another leaf pool here. So I removed it.